back/app/controllers/web_api/v1/projects_controller.rb

```diff
@@ -39,7 +42,7 @@ def index
 
     instance_options = {
       user_followers: user_followers,
-      timeline_active: TimelineService.new.timeline_active_on_collection(@projects),
+      timeline_active: TimelineService.new.timeline_active_on_collection(@projects.to_a),
```

Comment:
No, it doesn't look so:
```ruby
[1] pry(main)> puts Phase.where(project: Project.all).to_sql
SELECT "phases".* FROM "phases" WHERE "phases"."project_id" IN (SELECT "projects"."id" FROM "projects")
[2] pry(main)> puts Phase.where(project: Project.all.load).to_sql
2024-02-12 11:15:20.154257 D [8:8020 log_subscriber.rb:130] ActiveRecord::Base --   Project Load (0.9ms)  SELECT "projects".* FROM "projects"
SELECT "phases".* FROM "phases" WHERE "phases"."project_id" IN (SELECT "projects"."id" FROM "projects")
[3] pry(main)> puts Phase.where(project: Project.all.to_a).to_sql
2024-02-12 11:15:30.851167 D [8:8020 log_subscriber.rb:130] ActiveRecord::Base --   Project Load (1.3ms)  SELECT "projects".* FROM "projects"
SELECT "phases".* FROM "phases" WHERE "phases"."project_id" IN ('5cbf8d77-ed09-4a52-b5b0-a4c5ccf61fc5', '30932be0-2cc9-49b1-9ac0-b4e8a50dcb49', 'c2b88141-379f-4749-8768-67a7379ba27b')
```
Even if the projects are loaded (case 2), Rails still runs a subquery. It runs a separate query only when we pass an array, not an AR relation.

Reply:
> Shouldn't this be fast enough, as it's still one query?

But it needs to read two different tables, so AFAIU the amount of work is roughly the same whether it's one query with a subquery or two queries. But in this case, the subquery for projects was running 3 times instead of 1 we load projects separately. Hence the performance improvement from 3.2 to 1.7 seconds.

You can reproduce it if locally you go to Admin -> Reporting -> Create report -> Add "Visitor timeline" graph -> You'll see the spinner instead of the project select in master.
![image](https://github.com/CitizenLabDotCo/citizenlab/assets/91883434/ef071793-374b-4d31-87b1-5d46ab6908d4)

In this branch, you'll not see the spinner.

Reply:
I'm sorry, I'm I have a lot of work at the moment so I won't be able to try it out.

But how I understood, 2 queries should be a lot slower than one big query because of the overhead in communication between backend and database (a bit a similar principle as trying to have fewer API requests between frontend and backend). I think databases might also have their own caching mechanisms to deal with repetitions in queries, but not sure.

But since you report such a big improvement, it's really fine for me to merge. I just have trouble understanding how this part improves performance, as it's in contradiction with what I was taught.

Reply:
> I'm sorry, I'm I have a lot of work at the moment so I won't be able to try it out.

Yes, I understand. Thanks a lot for your time and for reviewing this PR!

I added this PR as an example to the next BE CDM. Let's discuss it in depth there.